backend/src/main/java/ca/bc/gov/hlth/mohums/util/FilterUserByOrgId.java

```diff
@@ -0,0 +1,89 @@
+package ca.bc.gov.hlth.mohums.util;
+
+import com.nimbusds.oauth2.sdk.util.CollectionUtils;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.function.Predicate;
+import net.minidev.json.JSONObject;
+import net.minidev.json.parser.JSONParser;
+import net.minidev.json.parser.ParseException;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionException;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+
+/**
+ *
+ */
+public final class FilterUserByOrgId implements Predicate<Object> {
+
+    private static final int ABBREVIATED_USER_ENTITY_LENGTH = 64;
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorizedClientsParser.class);
+
+    private final String orgId;
+    private final JSONParser jsonParser;
+    private final Expression expression;
+
+    public FilterUserByOrgId(final String id) {
+        this.orgId = id;
+        this.jsonParser = new JSONParser(JSONParser.DEFAULT_PERMISSIVE_MODE);
+        this.expression = new SpelExpressionParser()
+                .parseExpression("#this['attributes']['org_details']");
+    }
+    
+    Logger getLogger() {
+        return LOGGER;
+    }
+    
+    @Override
+    public boolean test(final Object userEntity) {
+        final EvaluationContext context = new StandardEvaluationContext(userEntity);
+
+        Collection<?> orgDetails = Collections.emptyList();
+
+        try {
+            orgDetails = expression.getValue(context, Collection.class);
+        } catch (final ExpressionException e) {
```

Comment:
Unless we know exactly why the exception is thrown and want to handle that case explicitly, we don't catch exceptions, we just re-throw them as unchecked exceptions, so `throw new RuntimeException(e)`.